tools/engine_tool/lib/src/commands/run_command.dart

```diff
@@ -76,26 +77,47 @@ final class RunCommand extends CommandBase {
     return null;
   }
 
-  String _selectTargetConfig() {
-    final String configName = argResults![configFlag] as String;
-    if (configName.isNotEmpty) {
-      return configName;
+  String _getDeviceId() {
```

Comment:
There are a couple of ways to test this. Maybe the simplest is to add tests like the `RunCommand` tests that already exist, and observe the `flutter` commands that are spawned as a result. Another option is to make these methods public, and unit test them by adding `@visibleForTesting`.

Reply:
Added a test.